app/services/mixtape_generator.rb

```diff
@@ -0,0 +1,32 @@
+class MixtapeGenerator
+  attr_reader :genre_name, :time
+
+  def initialize(genre_name:, time: 3600)
+    @genre_name = genre_name
+    @time = time
+  end
+
+  def self.perform(genre_name:, time: 3600)
+    new(genre_name: genre_name, time: time).perform
+  end
+
+  def perform
+    Mixtape.new(songs: pick_songs)
+  end
+
+  def pick_songs
+    Song.includes(:genre, :artist).where('songs.id in (?)', song_ids)
+  end
+
+  def song_ids
```

Comment:
I think you want to do this is the database as a `GROUP BY songs.id HAVING sum(songs.duration) > ?`, time

But will double check tonight